### ISSUE TITLE
Enemy Takes Damage (Player Hits Enemy)

### ISSUE THREAD

**Issue by WheeledParasite** [WheeledParasite/Hack-N-Slash#19]
## Summary
When the player "hits" the enemy, the enemy needs to take damage.  If the enemy's health drops to zero, it should be removed.

### Details

- The groundwork has already been laid for this in issue #18 where the "[EnemyBase](https://github.com/WheeledParasite/Hack-N-Slash/blob/main/Assets/Scripts/EnemyBase.cs#L35)" script was added.
- Based upon the weapon type, we need to reduce enemy health by the damage amount.
- If the enemy health drops to zero or less, enemy needs to die (be removed).  At some point, we will have a separate issue for enemy death animation, but for now, you can just remove the enemy from the scene.
- This will need to be coded in the DAMAGE observer class.
- Will need to wait until completion of issue #29

### Signal to Use
- OnDamage or OnHit

### Difficulty
- Easy-Intermediate
- This task will need to wait for issue #4 to be completed first because we will need to add weapons to the player first so that you can know how much damage is to be given to the enemy based upon weapon type.
- This task is partially dependent upon issue #16 as well.

**Comment by gururise** [comment]
Issue has been merged with #22.  Please see that issue instead. Closing this one.
